lib/libshare/os/linux/nfs.c

```diff
@@ -481,36 +489,39 @@ nfs_copy_entries(char *filename, const char *mountpoint)
 	size_t buflen = 0;
 	int error = SA_OK;
 
-	/*
-	 * If the file doesn't exist then there is nothing more
-	 * we need to do.
-	 */
 	FILE *oldfp = fopen(ZFS_EXPORTS_FILE, "r");
-	if (oldfp == NULL)
-		return (SA_OK);
-
 	FILE *newfp = fopen(filename, "w+");
 	fputs(FILE_HEADER, newfp);
-	while ((getline(&buf, &buflen, oldfp)) != -1) {
-		char *space = NULL;
-
-		if (buf[0] == '\n' || buf[0] == '#')
-			continue;
 
-		if ((space = strchr(buf, ' ')) != NULL) {
-			int mountpoint_len = strlen(mountpoint);
+	/*
+	 * The ZFS_EXPORTS_FILE may not exist yet. If that's the
+	 * case then just write out the new file.
+	 */
+	if (oldfp != NULL) {
+		while (getline(&buf, &buflen, oldfp) != -1) {
+			char *space = NULL;
 
-			if (space - buf == mountpoint_len &&
-			    strncmp(mountpoint, buf, mountpoint_len) == 0) {
+			if (buf[0] == '\n' || buf[0] == '#')
 				continue;
+
+			if ((space = strchr(buf, ' ')) != NULL) {
+				int mountpoint_len = strlen(mountpoint);
+
+				if (space - buf == mountpoint_len &&
+				    strncmp(mountpoint, buf,
+				    mountpoint_len) == 0) {
+					continue;
+				}
 			}
+			fputs(buf, newfp);
 		}
-		fputs(buf, newfp);
-	}
 
-	if (oldfp != NULL && ferror(oldfp) != 0) {
-		error = ferror(oldfp);
+		if (ferror(oldfp) != 0) {
+			error = ferror(oldfp);
+		}
+		fclose(oldfp);
```

Comment:
it looks like when closing newfp, we check for an error.  Should we do the same here?